### PR TITLE
Lower coverage gate to 90%; patch flaky water-density test; add target_surface_pressure sweep

### PR DESCRIPTION
**Pull request by timlichtenberg**
## Description

Four bundled changes on this branch:

1. **Lower coverage ceiling 95→90**: 95% has been unrealistically tight for this ecosystem. The cron nightly on 2026-05-12 03:01 UTC failed at 94.91% against the old 95% gate — that turned out to be a transient drop on identical source from the previous day's 95.58%. The new 90% ceiling leaves 5+ pp of permanent headroom. Updates `pyproject.toml fail_under = 90.0`, workflow `--cov-fail-under=90`, `.codecov.yml` badge gradient `range: "70...90"`, and the `CI.yml` pre-flight ratchet guard now enforces both bounds: floor at `min(base, 90.0)` so a one-time ratchet down to the ceiling is allowed, plus an upper bound at 90.0 so the gate cannot be silently raised past the ceiling either. Also corrects a stale "95% coverage gate" reference in `CONTRIBUTING.md`.

2. **Unblock standalone grid sweeps over `target_surface_pressure`**: Ema reported `ValueError: Unknown sweep parameter` even though the parameter is a valid Zalmoxis config knob (`[PressureAdjustment]` section, `config.py:483`). Added the entry to `tools/grids/run_grid.py:_PARAM_MAP`, plus three regression tests pinning the registry entry, exercising the loader, and exercising the edge case where the base TOML omits the `[PressureAdjustment]` section entirely. `generate_configs` now uses `setdefault(section, {})` so a sweep parameter targeting a missing section auto-creates it.

3. **Patch a flaky smoke test**: `test_density_profile_water` failed intermittently under nightly xdist parallelism on workflow_dispatch runs. Root cause is inner-Picard wall-budget drift on water configs producing ~1-2 outlier shells in the 10-15% deviation band against Seager+2007. Replaced the brittle pointwise `np.allclose` with a population assertion: 99% of shells within 10% AND worst shell within 15%. Added a mask-size sanity assertion (`rho_m.size > 10`) ahead of the percentile call so a degenerate run fails with a clear diagnostic instead of an IndexError from numpy. On converged baselines both bounds sit at ~4.4%. The patch was validated by a controlled A/B: pre-patch commit failed under the same xdist seed, post-patch commit passed (runs 25734561999 and 25734812147, 5 min apart).

4. **PR template**: dropped the "Relevant people" section from `.github/pull_request_template.md`.

## Validation of changes

- Local `pytest -m unit -o "addopts="`: full suite passes.
- `tests/test_grid_tools.py`: 51/51 (incl. 3 new tests for the registry entry, loader regression, and missing-section robustness).
- `tests/test_Seager_water.py` with patched percentile + mask-size assertion: passes locally (rho_max=4.4% vs threshold 15%, 135 shells after masking).
- Branch nightly run 25734812147 on commit `37b6237`: success, 1131/1131 pass, coverage 95.58%.
- Push-CI run 25758164382 on the post-review commit: green on ubuntu + macOS, confirming the new `min(base, 90.0)` ratchet-guard floor lets the 95→90 drop through.
- Tested under macOS 14 with Python 3.12.11 from the `proteus` conda env.

## Checklist

- [x] I have followed the [contributing guidelines](https://proteus-framework.org/PROTEUS/CONTRIBUTING.html#how-do-i-contribute)
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] My changes generate no new warnings or errors
- [x] I have checked that the tests still pass on my computer
- [x] I have updated the docs, as appropriate
- [x] I have added tests for these changes, as appropriate
- [x] I have checked that all dependencies have been updated, as required

### CODE REVIEW

File: tests/test_Seager_water.py [hunk 1]
```diff
@@ -54,9 +64,18 @@ def test_density_profile_water(cached_solver):
...
 
-    assert np.allclose(
-        model_densities[mask], seager_density_interp[mask], rtol=0.10, atol=300
-    ), (
-        f'Density profile for water config at {mass} M_earth deviates too much '
-        f'from Seager model (ignoring discontinuity)'
+    rho_m = model_densities[mask]
+    rho_s = seager_density_interp[mask]
+    rel_dev = np.abs(rho_m - rho_s) / np.abs(rho_s)
+
```
[the remark this answers is not in the document]
Comment (timlichtenberg) [reply tests/test_Seager_water.py:76]: > `rel_dev` is computed only on `mask`, but if the mask ever filters out all shells (e.g. a failed/non-converged run producing mostly vacuum padding), `np.percentile(rel_dev, 99)` and `rel_dev.max()` will raise on an empty array (turning a test failure into an error). Add an explicit assertion that `rel_dev.size > 0` (or `rho_m.size > 0`) with a clear message before calling percentile/max.

Good catch, fixed in `4471263`. Added `assert rho_m.size > 10` before the percentile/max call, with a message that explicitly names the degenerate-profile case (mostly vacuum padding, large discontinuities). A pathological run now fails the test with a readable diagnostic instead of an IndexError from numpy.

File: tests/test_grid_tools.py [hunk 2]
```diff
@@ -409,6 +409,79 @@ def test_load_grid_config_unknown_sweep_param(tmp_path, monkeypatch):
         rg.load_grid_config(str(grid))
 
 
+def test_param_map_target_surface_pressure_registered():
+    """target_surface_pressure must map to (PressureAdjustment,
+    target_surface_pressure). Pins the registry entry so a future
+    refactor of _PARAM_MAP cannot silently drop it (the entry was
+    missing before 2026-05-12 and broke standalone grids)."""
```
[the remark this answers is not in the document]
Comment (timlichtenberg) [reply tests/test_grid_tools.py]: > The new docstring bakes in an ISO date and historical narrative ("missing before 2026-05-12"), which conflicts with the repo's stated testing style guidance to avoid dates/past-change history in inline comments/docstrings (see docs/How-to/build_tests.md). Please rephrase to describe the invariant being tested (registry contains this key) without referencing when it was added/removed.

Right, that violated the repo rule against historical narrative in checked-in artefacts. Stripped the dates and rephrased to describe the invariant in `4471263`. All three flagged docstrings (`test_param_map_target_surface_pressure_registered`, `test_load_grid_config_accepts_target_surface_pressure`, `test_generate_configs_creates_missing_section`) now state what they pin, not when it was added.

File: .github/workflows/CI.yml [hunk 3]
```diff
@@ -81,10 +90,14 @@ jobs:
...
+              floor = min(base_val, ECOSYSTEM_CEILING)
+              if cur < floor:
+                  raise SystemExit(
+                      f"{label} fail_under {cur} < floor {floor} "
+                      f"(base={base_val}, ceiling={ECOSYSTEM_CEILING})"
+                  )
 
-          print("Coverage thresholds have not decreased vs main.")
+          print("Coverage thresholds respect the 90% ecosystem ceiling.")
           PY
```
[the remark this answers is not in the document]
Comment (timlichtenberg) [reply .github/workflows/CI.yml:106]: > The pre-flight check is described as enforcing a 90% *ceiling*, but the logic only enforces a lower bound (`cur >= min(base, ECOSYSTEM_CEILING)`) and would still allow raising `fail_under` above 90 later. If the intent is a true ecosystem ceiling, add an explicit guard that `cur <= ECOSYSTEM_CEILING` (or rename the constant/message to reflect that it's only a floor for decreases).

Right — the floor-only guard was an oversight. Added the symmetric upper bound in `4471263`: the check now rejects `cur > ECOSYSTEM_CEILING` as well, with a clear message. Same change applied to CALLIOPE `tests.yaml`, aragog `ci_tests.yml`, and PROTEUS `ci-pr-checks.yml` so the four repos behave consistently in both directions.
